llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -29981,7 +29981,8 @@ SDValue AArch64TargetLowering::LowerFixedLengthVECTOR_SHUFFLEToSVE(
           DAG, VT, DAG.getNode(Opc, DL, ContainerVT, Op1, Op1));
     }
 
-    if (Subtarget->hasSVE2p1() || Subtarget->hasSME2p1()) {
+    if (Subtarget->hasSVE2p1() ||
+        (Subtarget->hasSME2p1() && Subtarget->isSVEorStreamingSVEAvailable())) {
```

Comment:
Sorry, I should have said `isStreaming()`, i.e.
```suggestion
    if (Subtarget->hasSVE2p1() ||
        (Subtarget->hasSME2p1() && Subtarget->isStreaming())) {
```

Reply:
Actually, looking at the instruction description it seems your original code (using `isSVEorStreamingSVEAvailable`) was the right code, because with `+sme2p1` the instruction is also enabled in non-streaming mode. This can be seen from the pseudo-code for DUPQ, which has `IsSVEEnabled()`. This is actually something that @paulwalker-arm is trying to fix in other places (#145322).